libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/daemon/MessageBusListenerActor.scala

```diff
@@ -32,9 +36,11 @@ class MessageBusListenerActor[M](source: Source[M, NotUsed])(implicit messageLik
         log.warning("Listener already subscribed. Ignoring Subscribe message")
       case Failure(ex) =>
         log.error(ex, "Source/Listener died, subscribing again")
+        monitor.onError(ex)
         trySubscribeDelayed()
         context become idle
       case Done =>
+        monitor.onFinished
         log.info("Source finished, subscribing again")
         trySubscribeDelayed()
```

Comment:
We should not ignore this state, the question is if we should re-subscribe here...